### ISSUE TITLE
Remaining issues PEP 585 and 604

### ISSUE THREAD

**Issue by cdce8p** [PyCQA/pylint#4090]
Now that both MRs are done, I noticed some small remaining issues with the support for PEP 585 and PEP 604.

**Initial Merge Requests**
PEP 585: #4059
PEP 604: #4075

--

**Issue 585 - Missing error**
This is an error in Python **3.7** and **3.8**
```py
import typing

var: typing.Dict[int, list[int]]
```
Here the `__future__` import is missing. Currently this is not recognized as an error.

--

**Issue 604 - False positive**
This is allowed in Python **3.7+**
```py
from __future__ import annotations

var: dict[int, str | int]
```
`str | int` is allowed in type annotations if the `__future__` import is present. Currently the result is
```
E1131: unsupported operand type(s) for | (unsupported-binary-operation)
```

**Comment by q-wertz** [comment]
I still have the error with python 3.10 and pylint 2.14.3.

Type annotations are false positives… 

```
unsupported operand type(s) for | (unsupported-binary-operation)
```

**Comment by cdce8p** [comment]
@q-wertz Could you provide the code which is causing the issue? I just checked and AFAICT everything works as expected. Please also check if you have added `py-version` to your pylint config. If you did and it's set to `< 3.10`, the error is intentional.

**Comment by q-wertz** [comment]
Oh sorry. Stupid me. In the `.pylintrc` file was the wrong python version. Thanks for the hint and sorry for the false alarm.
